game/src/AI/States/Hooked.gd

```diff
@@ -0,0 +1,19 @@
+extends State
+"""
+This state is for enemy's response to being grappled with the hook.
+It simply waits until the player body collides with the hitbox, colors the body,
+ then transitions to `state_when_struck`
+"""
+
+
+export var state_when_struck: String
```

Comment:
Do we really want to type the path to the state in a variable? Does it ever need to be "hard-coded"? The case I can see it change is e.g. going to Stagger/Knockback if the enemy doesn't die with a given hit vs Die if it dies. But that's not something we should change with a property, I'd add that with the `msg` dictionary when the need arises